datafusion/expr/src/function.rs

```diff
@@ -66,22 +69,39 @@ pub struct AccumulatorArgs<'a> {
     ///
     /// If no `ORDER BY` is specified, `sort_exprs`` will be empty.
     pub sort_exprs: &'a [Expr],
+
+    /// Whether the aggregate function is distinct.
+    pub is_distinct: bool,
+
+    /// The input type of the aggregate function.
+    pub input_type: &'a DataType,
+}
+
+/// [`GroupsAccumulatorSupportedArgs`] contains information to determine if an
+/// aggregate function supports the groups accumulator.
+pub struct GroupsAccumulatorSupportedArgs {
```

Comment:
I know the current implementation of `groups_accumulator_supported` only needs these two fields, but I can see in the future needing more of them

Is there any reason we can't just pass `AccumulatorArgs` to `groups_accumulator_supported`?

Reply:
We could do this.